### PR TITLE
The Grand Stun Baton/Flash Double Nerf: Stamina Drain Edition

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Changes how many standard security weapons work. This is in an effort to make batons a little closer to traditional melee weapons, and retaining the powerful use flashes have as a self defense tool while removing their hard stun.

**Stun Batons**
They respect armor for their damage. It inflicts this damage against the chest and checks for chest armor.

The stun baton inflicts a delated burst of stamina damage after about a second from the hit. The baton also does less upfront damage but only has an attack cooldown of 1.5. This means you're hitting more frequently with your baton but doing less upfront damage. The baton also inflicts a stamina drain.

The stun effects of a stun baton is not negated by armor, but is partially stopped by stun resistance. The stamina drain cannot be removed except waiting out the 5 second duration. The upfront damage a stun baton inflicts is not negated by stun resistance.

Stun prods have an attack cooldown of 2.5 and after effect of 10 stamina damage.

**Classic Batons**
Only the contractor baton will cause a knockdown. All classic batons inflict confusion on the target. Because you smacked them on the head. Their damage is reduced by armor, much like the standard baton.

Contractor Batons have armor penetration which works for both their stamina damage and standard force. They are however weaker than before when attacking anyone in armor unless you're trying to use the baton as a lethal weapon, in which case it is stronger.

**Flashes**
These now cause a knockdown instead of a hard stun and also cause stamina drain. This is an 8 second knockdown. This means security would be better to start with flashing their target before moving to other weapons. 

Stun Resistance
Stun resistance prevents the stamina draining effect from flashes and the delayed stamina damage from batons. It does not affect the knockdown of flashes or the damage of batons.

## Why It's Good For The Game

Stun batons kind of invalidate melee weapons as a whole due to both doing stamina damage, disarming someone and also knocking them to the ground, which is usually a loss anyway because of shoves. Classic batons were no better either. But I really liked the idea of a delayed effect from being hit with a baton, so I have given them a different effect that still retains some of that disarming power.

This also makes stun resistance actually somewhat meaningful for stopping some of the nastier effects of stun batons without strictly making you immune to their effects.

Flashes were just kind of a gear check. They still ARE a gear check but on a much less severe level. They still can be useful as self defense tools. Their direct counters are still their counters.

## Changelog
:cl:
balance: Batons now respect armor as a whole.
balance: Stun batons now inflict a delayed hit of stamina damage after the initial hit. The baton also inflicts a stamina drain over time. This is an effective netgain on stamina damage per hit, but obviously is far slower for application.
balance: Stun batons have a lower cooldown but lower upfront stamina damage.
balance: Classic batons (like telebatons) confuse people on hit instead of knocking down.
balance: Contractor batons have armor penetration and retain the knockdown.
balance: Flashes cause an 8 second knockdown and stamina drain.
balance: Stun resistance protects from stun baton after effects entirely and remove stamina drain.
fix: Deleted copypasted code.
/:cl:

### CODE REVIEW

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -580,6 +567,20 @@
...
+	duration = 50
+	alert_type = null
+	status_type = STATUS_EFFECT_REFRESH
+
+/datum/status_effect/stamdrain/tick()
+	if(iscarbon(owner) && !HAS_TRAIT(owner, TRAIT_STUNRESISTANCE))
+		var/mob/living/carbon/H = owner
+		H.adjustStaminaLoss(3)
+	else
+		qdel(src)
```
Comment (Rohesie) [inline code/datums/status_effects/debuffs.dm:582]: Put this in its init instead of checking it every tick.

**Comment by ExcessiveUseOfCobblestone** [comment]
DMdoc your stuff smh!

**Comment by necromanceranne** [comment]
Tragic, I have to document all of Akrilla's work QwQ

**Comment by ExcessiveUseOfCobblestone** [comment]
i was just expecting you to doc your changed/new items but go ahead!

**Comment by necromanceranne** [comment]
I didn't add anything but I did touch pretty much all the batons so...

**Comment by wesoda25** [comment]
It looks like you're buffing stun batons, but I might have something wrong. I'll check out your changes on my local in a bit to check.

However I definitely dislike your changes to flashes. Knockdown is useful when getting rid of stuns but its a very lazy and inappropriate solution to flashes. You should make them simply focus on blinding (confusion would be good, maybe a slowdown or SMALL knockdown might be appropriate), and have it pierce through glasses, albeit with reduced effects.

**Comment by itseasytosee** [comment]
the baton changing from a near-immediate effect to a longer more impactful effect is a change that I would have to see situations played out to judge.
The changes to flashes don't seem to make sense. fluff-wise. the flash disorientates someone to the point of collapse and needs to recover for the 8 seconds. Now it knocks people presumably through the will of the sun god.

**Comment by wesoda25** [comment]
Ok. I played around a bit on my local and here's what I think. This post is sort of rambly but it helped me understand why these changes aren't so bad, so maybe it'll help other people (video good!). First and foremost - make sure aheal removes the shaky hand effect bc I don't think it did.

Anyways, these new stun batons are actually really fucking fun. They still are pretty useful due to the stamina damage they apply, and the shaky hands is a great way to make it difficult for people under its effects to fight back. One thing you might wanna consider though is just how effective armor is at negating their stamina damage. Works fine on stuff with rational armor values , like sec officers: 
aDxMEwg31v
**(Note: takes two hits to drop an unarmored opponent)**
but on things with a bit more:
w9deZRoBmx
So this would be a pretty significant buff to other armors and stuff, but tbh I think we've been working towards that so its fine. 

Looking at all this really though I feel as if you nerfed batons by taking away the knockdown, but buffed them by adding the shaky hand effect and continuous stamina - so if this really changes much aside from making armor more significant, I will be surprised. 

Still really don't like what you did with flashes though. I'd be willing to make some changes and then PR them to your branch or something for you to compare if you would like, since I think I have some good ideas about them.

**Comment by itseasytosee** [comment]
Look pretty neat. Knockdown after a moment didn't make too much sense anyway.
It's weaker overall against the slippery bastard trying to escape but stronger in a melee fight which I think is a solid change.

**Comment by necromanceranne** [comment]
@wesoda25 if you have some ideas, then sure I'm absolutely all ears. That said, I didn't want to leave security with completely zero options for knockdown, and while you might call it lazy, I feel like it's not a bad niche for the flash to fill.

**Comment by necromanceranne** [comment]
Just so you're in the know, convulsing is the same status effect defibs inflict upon you when they're emagged. So I guess we've come full circle.

**Comment by wesoda25** [comment]
> @wesoda25 if you have some ideas, then sure I'm absolutely all ears. That said, I didn't want to leave security with completely zero options for knockdown, and while you might call it lazy, I feel like it's not a bad niche for the flash to fill.

Its already filled though, thats what pepper spray does. Similarly flashbangs. 

I’m gonna play around with flashes tomorrow and report anything that seems balanced.

**Comment by Shadowflame909** [comment]
Telebatons are portable baseball bats

**Comment by Fikou** [comment]
i feel like classic batons have been made kinda useless now, you cant use them to get away from 1v1 encounters and confusing someone can just make them harder to hit

**Comment by Tlaltecuhtli** [comment]
you need 6 batonings on someone wearing basic armor? :joy: give sec knifes then as they seem more effective

**Comment by ATH1909** [comment]
> i feel like classic batons have been made kinda useless now, you cant use them to get away from 1v1 encounters and confusing someone can just make them harder to hit

Yeah, classic batons would be hit really, REALLY hard if this PR got merged, because confusing someone can make their movements less predictable and doesn't stop them from just shootin' ya. Maybe add a whip-like item-dropping effect to it?

**Comment by ATH1909** [comment]
Also, as a standard cyborg main (and someone who routinely grabs a flash for self-defense on most rounds), I'm going to have to give these flash changes a big YIKES.

They're my only way of doing stamina damage as a standard cyborg, and you've turned the stamina damage from them into a shitty stamina damage over time effect. Cyborgs can't shovestun to exploit the knockdown from flashes, and even though cyborgs have to flash someone like 7 times without their flash burning out!) over the course of around 35 seconds in order to FINALLY staminacrit them, they still get hardstunned by an opponent managing to get in ONE "melee" hit with a flash in response.

Please, god, replace the shitty 15 stamina damage over the course of 5 seconds (wait, actually, is it just 10 stamina damage over the course of 5 seconds, due to a tick being 2 seconds long?) effect with 35 immediate stamina damage or something, and maybe make flash burning out more forgiving to compensate. I don't want to have to resort to dragging Armsky around behind me everywhere in order to arrest murderboners/nonharmfully defend myself as a cyborg.

**Comment by ATH1909** [comment]
image

This seems like a pretty good alternate suggestion for the flash nerf.

**Comment by Megarop** [comment]
> The effects of the stun baton is a delayed series of convulsions that will result in you dropping items repeatedly, as well as taking some stamina damage over time. This is useful for keeping someone subdued for longer and encourages security officers to mix up their approach.

What does "mix up their approach" mean? Sec usually doesn't really have any other nonlethal weapons to combo with their stunbaton besides their disabler.

also batons making you drop whatever you're holding every 2 seconds while you're trying to fight sounds really annoying and not fun to deal with

**Comment by ATH1909** [comment]
image

It's good to know that necromanceranne truly cares about the cyborg experience.

**Comment by SomeoneYouProbablyKnow** [comment]
I'm all for nerfing flashes but they are like the only weapons borgs have. Consider giving borgs something in return or just giving flashes a shove as well for borgs to also engage in what is the combat system. And i'd like to see some change to the way borgs handle flash hits themselves

**Comment by AdamElTablawy** [comment]
Nerfing the only weapon borgs can use for self defense isn't great when the borg can be onehit by the same item.

**Comment by ATH1909** [comment]
> I'm all for nerfing flashes but they are like the only **nonlethal** weapons **for long term subduction** borgs have. Consider giving borgs something in return or just giving flashes a shove as well for borgs to also engage in what is the combat system. And i'd like to see some change to the way borgs handle flash hits themselves

made a correction for you there, so that necromanceranne doesn't dismiss this comment by listing off ways borgs can subdue you that all involve causing harm to you in some way

**Comment by necromanceranne** [comment]
For context, ATHATH is only concerned about whether he can play secborg as a standard borg, and if you were doing this before then I don't really have sympathy. We removed secborg for a reason. Silicons are not supposed to be cops. 

The flash is still, as of this current moment, good enough to break off a fight. Any fight you couldn't resolve before will still be unresolved after this change. The only thing you can't do is use the standard borg to arrest people with the flash.

@Tlaltecuhtli how quickly people have forgotten my katana wielding security force pr.

**Comment by SomeoneYouProbablyKnow** [comment]
I care more about subverted, emaged and malf borgs. And yes they have their hacked modules but an e sword isn't gonna help you much in a stun meta (arguably that's good but not when borgs need good stuns to not die to the same flashes). I'd honestly be fine with hacked status giving borg flashes some extra kick.

And fashes still being a hard stun on borgs just feels bad. why not make them very debilitating instead the same as a hard stun in effect but you'd feel like you have agency

**Comment by necromanceranne** [comment]
The fact that borgs are hardstunned by flashes is a separate issue to be entirely honest and something that could probably be addressed in a separate PR. I don't wanna touch that can of worms because right now borgs are nigh unstunnable short of those select weapons and suddenly taking that away from flashes cuts down one of a very tiny amount of tools available to stop them.

In fact if I'm being completely honest, borgs are kind of a hot mess combat wise. So much stuff just doesn't do anything to them it's obnoxious.

**Comment by Pugie** [comment]
Game mechanics that rng disarm you repeatedly aren't fun at all to be on the receiving end of and at the end of the day isn't that much different than baton knockdown, which results in you dropping your items first when you get knocked down and second when you get shoved - at least those two events can be accurately predicted.

I don't think it would be possible, but an alternative that comes to mind would be baton causing temporary paralysis in a random limb. Really anything that doesn't strip a fight response because without good armour it's looking like your only option is gonna be fight after getting hit once, and it's near impossible to fight with repeated item drop.

**Comment by tmtmtl30** [comment]
worried about what this will do to sleeping carp balance. bullet deflection means taking them down from afar is very tricky, and they have extra armor that makes a brute force approach unappealing. your best bet is stunbatons, but they have enough resistance to them to make you need to hit them several times (six sounds about right?). with this change, they'd become totally unstoppable armored.

probably something that could be fixed in a different PR, but it still worries me.

**Comment by ExcessiveUseOfCobblestone** [comment]
playing /tg/ and hating dice rolls for combat, smh!

**Comment by Coastfront** [comment]
The thing that angers me about this is this turns a bar of soap into a more effective flash, and it just pushes sec farther away from "normal" methods to stun people to now just dump 2 rubbershot shells into them. Also why do we need to make standard batons (mostly) completely useless at their designed purpose?

**Comment by Victor239** [comment]
Baton being affected by armour I like, how many hits is it now though for 1) standard officer armour 2) syndicate hardsuit and 3) cap hardsuit?

Flash changes seem totally unnecessary.

**Comment by Angustmeta** [comment]
Random forcedrops can be extremely stressful to new players. It's one of those things that cause painful brain syntax errors .

**Comment by senjx** [comment]
we are regressing back to the stun rng age and i love it

**Comment by wesoda25** [comment]
ok so yeah i looked at the code and am too lazy to figure this out but i think flashes would be better if:
1. can still do something to people with protection, just to lesser effect
2. Don't use a stun or knockdown, and if so its very short
3. make people pretty dizzy and give a longer blind
4. maybe give a short slowdown, but thats probably bad

This would make them actually good self defense weapons, something a crew member could use to escape a bad guy, regardless if that bad guy has flash protection. In turn its not the greatest tool for bad guys, and would be decent for sec for riots or something.

**Comment by necromanceranne** [comment]
It sure would but having a look at the code myself, holy fucking shit would this need an obscene amount of refactoring or snowflaking and frankly I cannot be arsed just to make the flashy toob slightly more useful against people without the protections.

I'll give it a bit of a think all the same about what to do about that.

**Comment by necromanceranne** [comment]
I feel like the convulsions, while interesting, were just as problematic as the knockdown. I've changed batons to a delayed burst of stamina damage and reduced their attack cooldown and delayed effect time. So you now hit more often with your batons but the damage is still slower and less upfront.

The benefits of the stamina drain is largely that it acts as an anti-stimulant of sorts at this point. Even against someone on chems, the stun baton will partially counteract several of the basic stamina heals. Which means hitting someone with it can makes your disabler more effective against someone deep in stimulants for a few seconds.

That's the intent anyway.

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (necromanceranne) [reply code/datums/status_effects/debuffs.dm]: Ended up changing the role of stun resistance a little so this might be unnecessary.

**Comment by ATH1909** [comment]
does the stamina drain still only last for 5 seconds?

**Comment by wesoda25** [comment]
Are you sure that its a good idea for batons to be less up front and more of a continuous thing? It would make dealing with criminals very difficult, I would think. Might be best to just have it apply 60 stamina damage flat, which is decreased based on armor. 

What I don't get is what are you trying to fix with these changes to batons? Your earlier changes seemed to change things up in a more interesting, but nonetheless similar way. Now it just seems like you're trying to make them pretty weak. I personally thought the issue with batons was their ability to stam crit easily, which is our new most accessible stun. I feel as if you aren't actually changing this.

**Comment by necromanceranne** [comment]
Stamcrit was only part of the issue. It's making one hit with the baton far less powerful overall and making it closer to a traditional weapon while still having some functional use for security and still being a respectable weapon for dealing with general tiders. The burst will still help with dealing with people in armor. The drain helps with dealing with stimulants. It's weaker but this is mostly attempts to soften nerf and still making it an informed choice for a given situation. 

Or, well, if you use it, it actually does something rather than nothing at all :U

**Comment by necromanceranne** [comment]
Actually I am still not satisfied with this. I think I'm approaching this problem in completely the wrong way. And that this is just not the right approach.

I think this needs to go even simplier. So I'll come back to this later in a new take.

For batons that is. I'm gonna splinter the flash nerf separately. It's good enough to handle rhe problem.

**Comment by ATH1909** [comment]
> Actually I am still not satisfied with this. I think I'm approaching this problem in completely the wrong way. And that this is just not the right approach.
> 
> I think this needs to go even simplier. So I'll come back to this later in a new take.
> 
> For batons that is. I'm gonna splinter the flash nerf separately. It's good enough to handle rhe problem.

image

**Pull request closed without merging**